sklearn/utils/_array_api.py

```diff
@@ -123,43 +137,36 @@ def get_namespace(*arrays):
     Returns
     -------
     namespace : module
-        Namespace shared by array objects.
+        Namespace shared by array objects. If any of the `arrays` are not arrays,
+        the namespace defaults to NumPy.
 
     is_array_api : bool
         True of the arrays are containers that implement the Array API spec.
     """
-    # `arrays` contains one or more arrays, or possibly Python scalars (accepting
-    # those is a matter of taste, but doesn't seem unreasonable).
-    # Returns a tuple: (array_namespace, is_array_api)
-
-    if not get_config()["array_api_dispatch"]:
-        return _NumPyApiWrapper(), False
+    return _get_namespace(
+        *arrays, array_api_dispatch=get_config()["array_api_dispatch"]
+    )
 
-    namespaces = {
-        x.__array_namespace__() if hasattr(x, "__array_namespace__") else None
-        for x in arrays
-        if not isinstance(x, (bool, int, float, complex))
-    }
 
-    if not namespaces:
-        # one could special-case np.ndarray above or use np.asarray here if
-        # older numpy versions need to be supported.
-        raise ValueError("Unrecognized array input")
+def _get_namespace(*arrays, array_api_dispatch=False):
```

Comment:
Please add a docstring to summarize the purpose of this helper and the meaning of the return values.